cas/ctg_store.c

```diff
@@ -997,28 +1001,53 @@ static int ctg_op_exec(struct m0_ctg_op *ctg_op, int next_phase)
 	struct m0_be_btree_cursor *cur    = &ctg_op->co_cur;
 	struct m0_be_tx           *tx     = &ctg_op->co_fom->fo_tx.tx_betx;
 	struct m0_be_op           *beop   = ctg_beop(ctg_op);
+	struct m0_cas_op          *cas_op = m0_fop_data(ctg_op->co_fom->fo_fop);
 	int                        opc    = ctg_op->co_opcode;
 	int                        ct     = ctg_op->co_ct;
+	uint64_t                   ver    = 0;
 	uint64_t                   zones;
+	bool                       use_ver = false;
 
 	zones = M0_BITS(M0_BAP_NORMAL) |
 		((ctg_op->co_flags & COF_RESERVE) ? M0_BITS(M0_BAP_REPAIR) : 0);
 
+	/*
+	 * Use the versioned btree API for ordinary CAS operations while
+	 * working in DTM0 environment.
+	 */
+	use_ver = ENABLE_DTM0 && ctg_is_ordinary(ctg_op->co_ctg);
```

Comment:
tis condition can be rewritten w.r.t. our oral disussion.

Reply:
yes, changed this thing as per our discussion